htlcswitch/link_test.go

```diff
@@ -5801,7 +5801,12 @@ func TestChannelLinkHoldInvoiceSettle(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Wait for Bob to receive the revocation.
+	// Wait for Alice to receive the revocation. This is needed
+	// because the settles are pipelined to the switch and otherwise
+	// the bandwidth won't be updated by the time Alice receives a
+	// response here.
+	time.Sleep(2 * time.Second)
```

Comment:
I'm a little confused about how this would work.  We can feed ticks to the `AckEventTicker`, but that won't help since we need to actually wait for the revocation from Bob (the duplicate settle doesn't matter here) and there doesn't seem to be a straightforward way to be notified upon receiving a revocation.  We could loop and timeout after some timeout interval, but that's not so clean either...

Reply:
I think we could actually *consume* the `AckEventTicker` tick because then that would let us know a revocation has been received.  Something like:
```
ticker := ctx.n.aliceServer.htlcSwitch.cfg.AckEventTicker
select {
case <-ticker.Ticks():
case <-time.After(20 * time.Second):
	t.Fatalf("alice did not receive revocation from bob")
}
```
Might be a bit hacky though? Either the test timeout would need to be bumped up or the mock interval for the ticker would need to be decreased.  If the revocation was received before this check and the tick was already consumed, it doesn't matter since we will get a duplicate tick.

Reply:
It seems `TestChannelLinkSingleHopPayment` fails for a similar reason and can be updated with this.

Reply:
Ah sorry, you're right. I see why the sleep is indeed needed, seems fine to me as is for now.

Reply:
Hm, the `TestChannelLinkSingleHopPayment` no longer fails on my machine with `flake-unit`, so maybe it's just my Mac.

Reply:
Nevermind, it flakes!

Reply:
Can use `lntest.WaitPredicate instead`?

Reply:
Get an import cycle then, but we could just copy the function?